ui/pages/confirmations/components/advanced-gas-fee-popover/advanced-gas-fee-gas-limit/advanced-gas-fee-gas-limit.js

```diff
@@ -26,8 +25,15 @@ const AdvancedGasFeeGasLimit = () => {
   const {
     gasLimit: gasLimitInTransaction,
     minimumGasLimitDec,
-    transaction: { chainId },
+    transaction: { chainId, originalGasEstimate },
   } = useGasFeeContext();
+  const originalGasEstimateDec =
+    originalGasEstimate !== undefined && hexToDecimal(originalGasEstimate);
+  const [maxGasLimit, setMaxGasLimit] = useState(
+    originalGasEstimateDec
+      ? Math.max(MAX_GAS_LIMIT_DEC, originalGasEstimateDec)
```

Comment:
If we're dynamically pushing the limit to what the estimate suggests, then do we need an upper bound at all?

Won't the UX still be awkward since the user won't be able to increase it and only reduce it?

Reply:
We update to what estimate suggest only if it is above `MAX_GAS_LIMIT_DEC`. IMHO `MAX_GAS_LIMIT_DEC` is `7920027` and is reasonable for any network.

The [issue](https://github.com/MetaMask/metamask-extension/issues/27089) has been reported in testnet, I donot expect any real use case of this high gas.